spark-cassandra-connector/src/main/scala/com/datastax/spark/connector/writer/BoundStatementBuilder.scala

```diff
@@ -38,12 +68,14 @@ private[connector] class BoundStatementBuilder[T](
       val converter = converters(i)
       val columnName = columnNames(i)
       val columnValue = converter.convert(buffer(i))
-      boundStatement.set(columnName, columnValue, CodecRegistryUtil.codecFor(columnTypes(i), columnValue))
```

Comment:
Although I had something else on my mind when I commented this part of code for the last time, I'm ok with this solution.

My idea was more or else to create two methods with identical signature:
```scala
def bindColumnLegacy(stmt: RichBoundStatement, idx: Int, value: AnyRef) ...
```
```scala
def bindColumnV4(stmt: RichBoundStatement, idx: Int, value: AnyRef) ...
```
Both of them would include the appropriate implementation which would replace the code block below:
```scala
if (columnValue == Unset || (ignoreNulls && columnValue == null)) {
  maybeLeaveUnset(boundStatement, columnName)
} else {
  val codec = CodecRegistryUtil.codecFor(columnTypes(i), columnValue)
  boundStatement.set(columnName, columnValue, codec)
}
```
with appropriately inlined `maybeLeaveUnset`.

Then, you would have to do:
```scala
val bindColumn: (RichBoundStatement, Int, AnyRef) => Unit = protocolVersion match {
  case pv if pv.toInt <= ProtocolVersion.V3.toInt => bindColumnLegacy
  case _ => bindColumnV4
}
```
somewhere in main class block.

Ultimately, the it would look like:
```scala
    rowWriter.readColumnValues(row, buffer)
    var bytesCount = 0
    for (i <- 0 until columnNames.size) {
      val converter = converters(i)
      val columnName = columnNames(i)
      val columnValue = converter.convert(buffer(i))
      bindColumn(boundStatement, i, columnValue)
      val serializedValue = boundStatement.getBytesUnsafe(i)
      if (serializedValue != null) bytesCount += serializedValue.remaining()
    }
```
where `bindColumn` method would be preselected on whole `BoundStatementBuilder` creation. Semantically it would be identical to creation of two subclasses of `BoundStatementBuilder` - each with different definition of that `bindColumn` method.